client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx

```diff
@@ -134,6 +143,23 @@ const PlansWrapper: React.FC< Props > = ( props ) => {
 		setDomain( freeDomainSuggestion );
 	};
 
+	/*
+	 * Could abstract this more by utilizing methods like
+	 * getPlanPersonalDetails().getNewsletterHighlightedFeatures() and
+	 * getPremiumPersonalDetails().getNewsletterHighlightedFeatures()
+	 * in @automattic/calypso-products, but would need more refactoring.
+	 */
+	const getHighlightedFeatures = () =>
+		isNewsletterFlow( flowName )
+			? [
+					FEATURE_CUSTOM_DOMAIN,
+					FEATURE_NEWSLETTER_IMPORT_SUBSCRIBERS_FREE,
+					FEATURE_UNLIMITED_SUBSCRIBERS,
+					FEATURE_PAYMENT_TRANSACTION_FEES_10,
+					FEATURE_PAYMENT_TRANSACTION_FEES_8,
+					FEATURE_PAYMENT_TRANSACTION_FEES_4,
+			  ]
+			: [];
```

Comment:
I don't like the idea of needing to just hard code features to highlight like this. In the comment here, I reference another method that is set to highlight features for plans comparison but that is a) by plan type and b) for the plans comparison specifically and c) does something different. But the fact that exists there, and we also need to highlight here, makes me think we could still use a more generic way to specific features to highlight, and pull that in when needed (like here and in the plans comparison code).

Reply:
Yeah, defining the features here seems misplaced. Just too many places where feature definitions/meta exist. It's also not going to be applied to `/plans/[ newsletter-site ]` (this might not be the intention right now, but imagine when we want these highlighted in admin too). Right now, there will also be a different order between `/setup/newsletter/plans` and `/plans/[ newsletter-site ]`

Using `@automattic/calypso-products` is one way of doing this (everyone's been using it for most things - like feature "footnotes", I think a similar concept). However, I find that location quite limiting for things that need to react to changes in the platform.

Have you considered creating a simple hook next to `npm-ready/data-store/use-plan-features-for-grid-plans.ts` and define these there, for now at least? As simple as:

```
useFeatureHighlights = ( { intent }: { intent: PlansIntent } ) => {
    if ( 'plans-newsletter' === intent ) {
        return ...
```

the rest would stay the same. and we can more easily refactor this later, as long as it's centralized somewhere 🤔

Reply:
I like the idea of consolidating it to a hook for now. At least that contains this logic in one place and, yes, makes it easy to find/add/adjust/refactor when needed.

I've added the hook.

And following a separate convo we had in slack, I've ensured that the hook will only return highlightedFeatures during signup (ie, when isInSignup is true).